Packs/ConcentricAI/Integrations/ConcentricAI/ConcentricAI.py

```diff
@@ -0,0 +1,630 @@
+        target['users-group'] = answer['permitted_grp_users']
+    if 'name' in answer and answer['name'] is not None:
+        target['file-name'] = answer['name']
+    if 'permitted_int_users' in answer and answer['permitted_int_users'] is not None:
+        target['user-internal'] = answer['permitted_int_users']
+    if 'permitted_orphans' in answer and answer['permitted_orphans'] is not None:
+        target['users-orphans'] = answer['permitted_orphans']
+    return target
+
+
+def transform_file_information(target: dict, risk_dict: dict, queryClient: QueryClient, loginClient: LoginClient):
+
+    if target['risk_id'] is not None:
+        target['risk_names'] = get_rule_names(target['risk_id'], risk_dict)
+    if 'risk_id' in target:
+        target.pop('risk_id')
+    if 'category' in target:
+        target.pop('category')
+    if 'duplicate_contrib_cids' in target:
+        target.pop('duplicate_contrib_cids')
+    if 'word_cloud' in target:
+        target.pop('word_cloud')
+    if 'confidence' in target:
+        target.pop('confidence')
+    if 'ownerDetails' in target:
+        if target['ownerDetails'] is not None:
+            target['ownerDetails'] = target['ownerDetails']['name']
+    if 'near_duplicate_contrib_cids' in target:
+        if target['near_duplicate_contrib_cids'] is not None:
+            file_names: list = []
+            for cid in target['near_duplicate_contrib_cids']:
+                res = queryClient.get_file_names(loginClient, cid)
+                if res is not None:
+                    path = res['path']
+                    name = res['name']
+                    file_names.append(path + " --> " + name)
+            target['near_duplicate_files'] = file_names
+            target.pop('near_duplicate_contrib_cids')
+    return target
+
+
+def test_module(client: LoginClient):
+    token = client.fetch_api_token()
+    if token:
+        return 'ok'
+
+
+def fetch_incidents(loginClient: LoginClient, queryClient: QueryClient, last_run: Dict[str, int], max_results: int):
+    global SCROLL_ID_INCIDENT
+    last_fetch = last_run.get('last_fetch', None)
+    scroll_id = last_run.get('scroll_id', None)
+    if last_fetch is None:
+        from_time = 0
+        to_time = int(datetime.now().timestamp() * 1000)
+        last_fetch = int(to_time)
+        to_time = last_fetch
+    else:
+        last_fetch = int(last_fetch)
+        from_time = last_fetch
+        to_time = int(datetime.now().timestamp() * 1000)
+        SCROLL_ID_INCIDENT = str(scroll_id)
+
+    min_severity = demisto.getParam('min_severity')
+    if min_severity is None:
+        min_severity = 'low'
+
+    flag = True
+    newAlerts = True
+    pageIndex = 1
+    answers: list = []
+
+    total_count = 0
+    max_records = None
+
+    while flag is True:
+        res = queryClient.fetch_incidents(min_severity, str(from_time), str(to_time), str(pageIndex), loginClient)
```

Comment:
The idea of the fetch looks good but I think we might miss some incidents in the current implementation:
If there are more than 300 incidents to fetch then once we hit 300 we break out of this loop and the next fetch will start from a new start time. Meaning that we lose those incidents. Isn't that right?
I suggest 2 things: 
1. Keep in the `IntegrtaionContext` also a field `pageIndex` which will contain the page index that way we can in the next fetch keep fetching from that page. When doing this we probably wouldn't want to change the `start_time` and `end_time` so the query will stay the same.
2. I saw that in `queryClient.fetch_incidents` you hardcode `pageSize: 150`, I suggest changing this to be `MAX_INCIDENTS_TO_FETCH`. That way we don't have to loop in a while loop but do only one HTTP request.

Reply:
@orlichter1 , now I won't miss those incidents. I retrieve em by storing scroll id. 
Answer on point-1 ) The pageIndex really doesn't matter much , since I keep track of incidents using scroll_id.
Answer on point-2) would set pageSize to max_fetch , that we I can avoid looping !

Reply:
What's the difference between `pageIndex` and `scroll_id` then?
I'll be more than happy to talk about this in slack or a quick zoom meeting 😄 
You can find me in the DFIR channel (Or Lichter)

Reply:
removed storing pageIndex and the looping logic.